src/Castle.Core/Components.DictionaryAdapter/Xml/Internal/Cursors/XPath/XPathCompiler.cs

```diff
@@ -441,7 +441,7 @@ private static bool IsWhitespace(char c)
 
 			private static bool IsNameStartChar(char c)
 			{
-#if DOTNET40
+#if DOTNET45
 			    return XmlConvert.IsStartNCNameChar(c);
```

Comment:
Might want to introduce a new conditional compilation symbol `FEATURE_XMLCONVERT`, since we're shortly going to introduce .NET Standard 2.x targets and those will support this API, too.